frontend/src/metabase/visualizations/components/TableInteractive.jsx

```diff
@@ -748,15 +802,71 @@ export default class TableInteractive extends Component {
     );
   };
 
+  getDisplayColumnWidth = ({ index: displayIndex }) => {
+    if (this.state.IDColumn && displayIndex === 0) {
+      return SIDEBAR_WIDTH;
+    }
+
+    // if we have an ID column, we've added a column of empty cells and need to shift
+    // the display index to get the data index
+    const dataIndex = this.state.IDColumn ? displayIndex - 1 : displayIndex;
+
+    return this.getColumnWidth({ index: dataIndex });
+  };
+
   getColumnWidth = ({ index }) => {
     const { settings } = this.props;
     const { columnWidths } = this.state;
     const columnWidthsSetting = settings["table.column_widths"] || [];
+
     return (
       columnWidthsSetting[index] || columnWidths[index] || MIN_COLUMN_WIDTH
     );
   };
 
+  handleHoverRow = (event, rowIndex) => {
+    const hoverDetailEl = this.detailShortcutRef.current;
+
+    if (!hoverDetailEl) {
+      return;
+    }
+
+    const detailButton = hoverDetailEl.querySelector(
```

Comment:
To avoid mixed approach and completely eliminate `querySelector`, you can add `buttonRef` prop to `DetailShortcut` and use two separate refs in the parent component

Reply:
instead of juggling multiple refs, I restructured the css a bit to get the same behavior by only manipulating one element.